File: plugin/zeppelin/src/main/scala/jetbrains/zeppelin/utils/dependency/LibraryDescriptor.scala
package jetbrains.zeppelin.utils.dependency

import com.intellij.openapi.roots.OrderRootType
import com.intellij.openapi.roots.libraries.NewLibraryConfiguration
import com.intellij.openapi.roots.ui.configuration.libraryEditor.LibraryEditor
import jetbrains.zeppelin.wizard.ZeppelinLibraryDescription.LIBRARY_CAPTION

case class LibraryDescriptor(version: String,
                             classes: List[String],
                             sources: List[String] = List(),
                             docs: List[String] = List()) {
  def createNewLibraryConfiguration: NewLibraryConfiguration = {
    new NewLibraryConfiguration(LIBRARY_CAPTION + "_" + version) {
      override def addRoots(editor: LibraryEditor): Unit = {
        classes.foreach(it => {
          editor.addRoot(it, OrderRootType.CLASSES)
        })

        sources.foreach(it => {
          editor.addRoot(it, OrderRootType.SOURCES)
        })

        docs.foreach(it => {
          editor.addRoot(it, OrderRootType.DOCUMENTATION)
        })
      }
    }
  }
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/dependency/LibraryDescriptor.kt
package org.intellij.plugin.zeppelin.dependency.dependency

import com.intellij.openapi.roots.OrderRootType
import com.intellij.openapi.roots.libraries.NewLibraryConfiguration
import com.intellij.openapi.roots.ui.configuration.libraryEditor.LibraryEditor
import org.intellij.plugin.zeppelin.idea.wizard.ZeppelinLibraryDescription

data class LibraryDescriptor(val version: String,
                             val classes: List<String>,
                             val sources: List<String> = listOf(),
                             val docs: List<String> = listOf()) {
    fun createNewLibraryConfiguration(): NewLibraryConfiguration {
        return object : NewLibraryConfiguration(ZeppelinLibraryDescription.LIBRARY_CAPTION + "_" + version) {
            override fun addRoots(editor: LibraryEditor) {
                classes.forEach { it -> editor.addRoot(it, OrderRootType.CLASSES) }
                sources.forEach { it -> editor.addRoot(it, OrderRootType.SOURCES) }
                docs.forEach { it -> editor.addRoot(it, OrderRootType.DOCUMENTATION) }
            }
        }
    }
}